src/events/app/index.ts

```diff
@@ -126,42 +126,46 @@ export async function handleAppInstallOnRepoEvent(
 }
 
 /**
- * General handling for syncing repositories.
+ * General handling for syncing repositories of all owners.
  * @param repoKeys
- * @param github
- * @param log
+ * @param app
  * @param pullService
  * @param issueService
  * @param commentService
  * @param contributorService
  */
 async function handleSyncRepos(
   repoKeys: RepoKey[],
-  github: InstanceType<typeof ProbotOctokit>,
-  log: Logger,
+  app: Probot,
```

Comment:
Maybe we only need a github client not whole app?

Reply:
The logger instance is also needed.

Reply:
Can we get the logger from context?

Reply:
Yes, we can. But maybe we also need the `app.auth` API, instead of a github client.

Reply:
Could you please explain why?

Reply:
Because we need to call `app.auth` to finish authorization according to installation id when we traverse repokey.

![image](https://user-images.githubusercontent.com/5086433/108324137-7a29e880-7202-11eb-8960-a910dd55e231.png)

Reply:
Sorry, I miss it 😆